### PR TITLE
Convert job sequence results partial to component

### PR DESCRIPTION
**Pull request by jonatanschroeder**

### CODE REVIEW

File: apps/prairielearn/src/lib/server-jobs.types.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  jobs: JobWithToken[];
+};
+
+export type JobWithFormattedOutput = Omit<JobWithToken, 'output'> & {
+  output: string;
+  output_raw: Job['output'];
+};
+export type JobSequenceWithFormattedOutput = Omit<JobSequenceWithTokens, 'jobs'> & {
+  jobs: JobWithFormattedOutput[];
+};
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/server-jobs.types.ts:35]: In theory, the "with formatted output" could be removed in place of formatting directly in the template, right? Same with token generation?

**Review by nwalters512 — APPROVED** [review]
Tested locally, this looks good to me.

File: apps/prairielearn/src/lib/server-jobs.types.ts (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/server-jobs.types.ts]: Indeed, done in 65af26c. For tokens I'd prefer not to move them as that might create some dependency nightmares between assets and server-side code.

I also converted the formatted date stuff in 890874f. Do you want to review those changes?

Comment (nwalters512) [reply apps/prairielearn/src/lib/server-jobs.types.ts]: > For tokens I'd prefer not to move them as that might create some dependency nightmares between assets and server-side code.

Is your concern that we'd eventually want to use the job sequence templates directly on the client?

I'll take a look at your most recent changes now.

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/lib/server-jobs.types.ts (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/server-jobs.types.ts]: > Is your concern that we'd eventually want to use the job sequence templates directly on the client?

Reviewing what I have in mind, it's not actually that bad of an issue. Currently the jobSequenceResults.ts asset imports a type from the JobSequenceResults.html.ts component. If tokens were generated in the component, this type would need to go to a different types.ts file.

That said, I still have a bit of reservations about where the line lies between computations allowed or not done in an html.ts component/page. Anything related to formatting is fair game, but once we start adding calculation of tokens this starts to move the line a bit. Not a big deal, but I assume we'd want to keep some of this formatting content clean of this kind of calculation.

**Pull request merged**